darts/dataprocessing/transformers/window_transformer.py

```diff
@@ -0,0 +1,691 @@
+
+                series_subset += [
+                    (s.univariate_component(c), transformation, builtins)
+                    for (s, c) in itertools.product(series, transformation["comp_id"])
+                ]
+
+            else:
+                # apply the transformation to a specific component in a specific series
+                # if a different component is provided for each selected series
+                if all(isinstance(x, list) for x in transformation["comp_id"]) and len(
+                    transformation["comp_id"]
+                ) == len(transformation["series_id"]):
+                    # test that components exist in the corresponding series
+                    raise_if_not(
+                        all(
+                            max(c_idxvec) <= series[s_idx].width - 1
+                            for (s_idx, c_idxvec) in zip(
+                                transformation["series_id"], transformation["comp_id"]
+                            )
+                        ),
+                        "Some components are not available in the provided series.",
+                    )
+                    series_subset += [
+                        (
+                            series[s_idx].univariate_component(c_idx),
+                            transformation,
+                            builtins,
+                        )
+                        for (s_idx, c_idx) in itertools.chain(
+                            *[
+                                list(itertools.product([s_idvec], c_idvec))
+                                for (s_idvec, c_idvec) in zip(
+                                    transformation["series_id"],
+                                    transformation["comp_id"],
+                                )
+                            ]
+                        )
+                    ]
+                # if the same components are provided for all the selected series
+                else:
+                    # test that the components exist in each series
+                    raise_if_not(
+                        all(
+                            c_idx <= series[s_idx].width - 1
+                            for s_idx in transformation["series_id"]
+                            for c_idx in transformation["comp_id"]
+                        ),
+                        "Some components are not available in the provided series.",
+                    )
+                    series_subset += [
+                        (
+                            series[s_id].univariate_component(c_id),
+                            transformation,
+                            builtins,
+                        )
+                        for (s_id, c_id) in itertools.product(
+                            transformation["series_id"], transformation["comp_id"]
+                        )
+                    ]
+
+        return iter(series_subset)  # the iterator object for ts_transform function
+
+    def ts_transform(
```

Comment:
could be nice to add the `@staticmethod` decorator for clarity here too

Reply:
How about implementing most of this logic in a `TimeSeries` function - e.g. `TimeSeries.window_transform()` ? That could be nice because it would allow users to obtain transformed series very simply without even instantiating a transformer (similar to how they can call `TimeSeries.map()` to apply a function on the series). WDYT?

Reply:
yes, can do

Reply:
logic moved to TimeSeries.window_transform()